android/engine/src/main/java/org/smartregister/fhircore/engine/ui/register/BaseRegisterActivity.kt

```diff
@@ -710,7 +712,12 @@ abstract class BaseRegisterActivity :
       showToast(getString(R.string.session_expired))
       accountAuthenticator.logout()
     } else {
-      showToast(getString(R.string.sync_failed))
+      if (exceptions.map { it.exception }.filterIsInstance<UnknownHostException>().isNotEmpty() ||
```

Comment:
think you can avoid looping through the exceptions list twice, by using a different closure, and you can avoid the `isNotEmpty` check by using `any`, something like

```kotlin
....any { exception: exception.isInstance(UnknownHostException) ||
                     exception.isInstance(InterruptedIOException) }
...
```

just writing that out, I think this deserves to be made a boolean utility function, like `containsAnyInstances([list of items to check], [list of classes])`

that returns true is anything in the list of items is of any of the classes in the list of classes, and false otherwise

Reply:
I have refactored to incorporate this. 
I'm not sure where would be ideal to place the `containsAnyInstances` function. 
cc @ellykits

Reply:
I have refactored to incorporate this. 
I'm not sure where would be ideal to place the `containsAnyInstances` function. 
cc @ellykits